src/main/java/codeu/controller/GeneratingData.java

```diff
@@ -0,0 +1,133 @@
+    
+	private String[] users = new String[]{"boss", "manager", "ted", "frank", "linda", "exuser"};
+
+
+    /** Set up state for handling chat requests. */
+    @Override
+    public void init() throws ServletException {
+      super.init();
+      setConversationStore(ConversationStore.getInstance());
+      setMessageStore(MessageStore.getInstance());
+      setUserStore(UserStore.getInstance());
+    }
+    
+    void setConversationStore(ConversationStore conversationStore) {
+        this.conversationStore = conversationStore;
+    }
+    
+    void setMessageStore(MessageStore messageStore) {
+        this.messageStore = messageStore;
+    }
+    
+    void setUserStore(UserStore userStore){
+    	this.userStore = userStore;
+    }
+    
+      
+	/**
+	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+	 */
+	public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		ServletContext cntxt = getServletContext();
+		FileInputStream fstream = 
+                new FileInputStream(getServletContext().getRealPath("/datafile"));	
+		
+		//six mock users - will randomly be assigned a message
+		for(String user: users){
+		  newUser(user);
+		}
+		
+		//one convo for now - can expand later
+		String conversation = "codeu_faq";
+
+	    newPrivateConvo("boss", "codeu_faq");
+	    
+		BufferedReader br = new BufferedReader((new InputStreamReader(fstream)));
+		String line = br.readLine();
+		List<String> messages = new ArrayList<String>();
+		while(line !=null){
+			messages.add(br.readLine());
+			line = br.readLine();
+		}
+		Random rand = new Random();
+		for(int i = 0; i <20; i++){
+		  newMessage(users[rand.nextInt(5)], conversation, messages.get(rand.nextInt(38)));
+		}
+	  
+	}
+	
+	public void newUser(String username){
+		String hashed =BCrypt.hashpw("1234", BCrypt.gensalt());
+		User newUser = new User(UUID.randomUUID(), username , hashed, Instant.now());
+	    userStore.addUser(newUser);
+	}
+	public void newConvo(String username, String title){
+		User user = userStore.getUser(username);
+		Conversation newConvo =
+		            new Conversation(UUID.randomUUID(), user.getId(), title, Instant.now());
+		conversationStore.addConversation(newConvo);
+	}
+	public void newPrivateConvo(String username, String title){
+		User user = userStore.getUser(username);
+		Conversation newConvo =
+		            new Conversation(UUID.randomUUID(), user.getId(), title, Instant.now(), true);
+		for(String name: users){
+		  User users = userStore.getUser(name);
+		  newConvo.addUserToWhitelist(users);
+		}
+		conversationStore.addConversation(newConvo);
+	}
+	public void newMessage(String username, String title, String message){
+		User user = userStore.getUser(username);
+		Conversation convo = conversationStore.getConversationWithTitle(title);
+		Message newMessage = new Message(UUID.randomUUID(), convo.getId(), user.getId(), message, Instant.now());
```

Comment:
To test the chat frequency at a given time, could we have some randomness regarding the `Instant` for the messages?
I know it'd be a bit weird to create messages in the past and future but it would help with testing.

Reply:
I'll work on the right now!